### ISSUE TITLE
[BUG] _line_stack() doesn't accept np.ndarray in stackers argument

### ISSUE THREAD

**Issue by harabat** [bokeh/bokeh#10067]
**Description of problem**
`varea_stack()` and `vline_stack()` process the `stackers` argument differently: the former takes `list`, `np.ndarray`, while the latter only takes `list`. This also holds for `harea_stack()` and `hline_stack()`. 

**Possible solutions**
For `_line_stack()`, doing `stackers=list(stackers)` in `plotting/figure.py` fixes the issue.

**Additional context**
```
np.random.seed(seed=10)
df = pd.DataFrame(np.random.randint(10, 100, size=(10, 10))).add_prefix('y')
p = figure(
    x_range=(0, len(df)-1),
    y_range=(0, 800),
    height=300, 
    width=300
)
```

1. `varea_stack()` with a `pandas.core.indexes.base.Index`
```
p.varea_stack(
    stackers=df.columns,
    x='index', 
    color=brewer['Spectral'][10], 
    source=df
)
show(p)
```
![image](https://user-images.githubusercontent.com/45575009/82749879-28a01a80-9da4-11ea-98fa-99113da92585.png)

2. `vline_stack()` with a `pandas.core.indexes.base.Index`
```
p.vline_stack(
    stackers=df.columns,
    x='index', 
    color=brewer['Spectral'][10], 
    source=df
)
show(p)
```
```
RuntimeError: 

Expected y to reference fields in the supplied data source.

When a 'source' argument is passed to a glyph method, values that are sequences
(like lists or arrays) must come from references to data columns in the source.
```

3. `varea_stack()` with `list`
```
p.varea_stack(
    stackers=list(df.columns),
    x='index', 
    color=brewer['Spectral'][10], 
    source=df
)
show(p)
```
![image](https://user-images.githubusercontent.com/45575009/82749996-1d99ba00-9da5-11ea-980e-c10bc557b72a.png)

4. `vline_stack()` with `list`
```
p.vline_stack(
    stackers=list(df.columns),
    x='index', 
    color=brewer['Spectral'][10], 
    source=df
)
show(p)
```
![image](https://user-images.githubusercontent.com/45575009/82749991-0bb81700-9da5-11ea-97e5-75b735a0ce3f.png)

**Comment by bryevdv** [comment]
@harabat I think the ultimate root cause is this type check in `Figure._line_stack`:

https://github.com/bokeh/bokeh/blob/9d160515bdcd4e0a2d8c937a68e3cdaebe4d5c4b/bokeh/plotting/figure.py#L1330

(and another similar one a few line below as well). 

It think it would be preferable to try and address this root cause. The `_line_stack` is re-used by both `vline_stack` and `hline_stack` and the purpose of the type check is to try and determine which of `x` or `y` has the sequence of stackers. Some options might be:

* widen the type checks to include any sequence types (e.g. arrays and Series)
* try checking for "not string" instead
* have `vline_stack` and `hline_stack` pass in an explicit flag that indicates which block should run, so that the type checks can be avoided altogether

**Comment by aastha-shah** [comment]
Hi, I'm interested in taking this up. However, I'm new to OSS contributions; I couldn't find this on your 2.1 milestone, so am I right in assuming there's no due date for this? Thanks!

**Comment by bryevdv** [comment]
@aastha-shah Correct, we don't put `Good First Issue` issues in any milestone, so that they can be taken up at any time by a new contributor.

**Comment by aastha-shah** [comment]
@bryevdv Hi, so I was thinking of incorporating your first two suggestions and simply checking if the object has the `__getitem__` attribute and if it isn't a `str` or `dict`. Also, would your Slack be a good place to clarify technical qs about the contributing process? Thanks.

**Comment by harabat** [comment]
Hey @aastha-shah, are you still working on this? I'm the one who opened this issue, willing to implement it unless you're doing it.

**Comment by bryevdv** [comment]
@aastha-shah sorry your question got lost in all things going on for the release. Yes the slack is a good place for development-related chat https://slack-invite.bokeh.org

@harabat Bokeh is an open, volunteer project. Occasionally core devs will self-assign an issue, and I suppose if someone asked to be assigned, we could/would do that. But in general, anyone should feel free to submit any PR at any time.

**Comment by harabat** [comment]
@bryevdv given that aastha-shah said they were new to OSS, wouldn't want my going ahead without checking in with them risk being perceived as a lack of good manners and alienate them from contributing further. 

@aastha-shah I'll start work on this next week, unless you want to have a go

**Comment by aastha-shah** [comment]
@harabat extremely apologetic about missing this. I had worked on this but then I ran into a bunch of issues with conda on my device and I was unable to test it, I should've mentioned it here. Please go ahead (would appreciate it if you had resources on contributing to OSS as well!)

**Comment by mosc9575** [comment]
I think it is enough to change

1. 

https://github.com/bokeh/bokeh/blob/9d160515bdcd4e0a2d8c937a68e3cdaebe4d5c4b/bokeh/plotting/figure.py#L1325
to `if isinstance(x, str) and isinstance(y, str):`
2.
https://github.com/bokeh/bokeh/blob/9d160515bdcd4e0a2d8c937a68e3cdaebe4d5c4b/bokeh/plotting/figure.py#L1330

to `if isinstance(x, str):`

and
3.

https://github.com/bokeh/bokeh/blob/9d160515bdcd4e0a2d8c937a68e3cdaebe4d5c4b/bokeh/plotting/figure.py#L1336

to `if isinstance(x, str):`


This would enable to plot pandas DataFrames, too. Pandas Series won't work, which makes sense to me, because if you have only one line, there is no reson to stack anything.

**Comment by bryevdv** [comment]
@mosc9575 If you would like to submit a PR please feel free, we will need to discuss how to add some minimal tests
